grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcDecoratingServiceTest.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.grpc.GrpcSerializationFormats;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleRequest;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleResponse;
+import com.linecorp.armeria.grpc.testing.Metrics.GaugeRequest;
+import com.linecorp.armeria.grpc.testing.Metrics.GaugeResponse;
+import com.linecorp.armeria.grpc.testing.MetricsServiceGrpc.MetricsServiceBlockingStub;
+import com.linecorp.armeria.grpc.testing.MetricsServiceGrpc.MetricsServiceImplBase;
+import com.linecorp.armeria.grpc.testing.ReconnectServiceGrpc.ReconnectServiceBlockingStub;
+import com.linecorp.armeria.grpc.testing.ReconnectServiceGrpc.ReconnectServiceImplBase;
+import com.linecorp.armeria.grpc.testing.TestServiceGrpc.TestServiceBlockingStub;
+import com.linecorp.armeria.grpc.testing.TestServiceGrpc.TestServiceImplBase;
+import com.linecorp.armeria.protobuf.EmptyProtos.Empty;
+import com.linecorp.armeria.server.DecoratingHttpServiceFunction;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.annotation.Decorator;
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import io.grpc.stub.StreamObserver;
+
+class GrpcDecoratingServiceTest {
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension(true) {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.requestTimeoutMillis(5000);
+            sb.decorator(LoggingService.newDecorator());
+            sb.service(GrpcService.builder()
+                                  .addService(new TestServiceImpl())
+                                  .addService(new MetricsServiceImpl())
+                                  .addService(new ReconnectServiceImpl())
+                                  .build());
+        }
+    };
+
+    private static String FIRST_TEST_RESULT = "";
+    private static String SECOND_TEST_RESULT = "";
+    private static String THIRD_TEST_RESULT = "";
+
+    @Test
+    void methodDecorators() {
+        final TestServiceBlockingStub client = Clients.builder(server.httpUri(GrpcSerializationFormats.PROTO))
```

Comment:
Use `GrpcClients.builder()`?
```suggestion
        final TestServiceBlockingStub client = GrpcClients.builder(server.httpUri())
```